No enum constant com.flowcrypt.email.service.actionqueue.actions.Action.Type.REGISTER_USER_PUBLIC_KEY

```
[SUMMARY]
java.lang.IllegalArgumentException: No enum constant com.flowcrypt.email.service.actionqueue.actions.Action.Type.REGISTER_USER_PUBLIC_KEY
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:


[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:30)
        at com.flowcrypt.email.jetpack.workmanager.ActionQueueWorker.doWork(ActionQueueWorker.kt:46)
        at com.flowcrypt.email.jetpack.workmanager.ActionQueueWorker$doWork$1.invokeSuspend(Unknown Source:14)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.lang.IllegalArgumentException: No enum constant com.flowcrypt.email.service.actionqueue.actions.Action.Type.REGISTER_USER_PUBLIC_KEY
        at java.lang.Enum.valueOf(Enum.java:259)
        at com.flowcrypt.email.service.actionqueue.actions.Action$Type.valueOf(Unknown Source:2)
        at com.flowcrypt.email.util.google.gson.ActionJsonDeserializer.deserialize(ActionJsonDeserializer.kt:35)
        at com.flowcrypt.email.util.google.gson.ActionJsonDeserializer.deserialize(ActionJsonDeserializer.kt:26)
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
        at com.google.gson.Gson.fromJson(Gson.java:1058)
        at com.google.gson.Gson.fromJson(Gson.java:1016)
        at com.google.gson.Gson.fromJson(Gson.java:959)
        at com.google.gson.Gson.fromJson(Gson.java:927)
        at com.flowcrypt.email.database.entity.ActionQueueEntity.toAction(ActionQueueEntity.kt:34)
        at com.flowcrypt.email.jetpack.workmanager.ActionQueueWorker.doWork(ActionQueueWorker.kt:42)
        ... 7 more
```

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/service/actionqueue/actions/Action.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/service/actionqueue/actions/Action.kt
@@ -29,6 +29,7 @@ interface Action : Parcelable {
    * This class contains information about all action types.
    */
   enum class Type constructor(val value: String) : Parcelable {
+    NONE("none"),
     BACKUP_PRIVATE_KEY_TO_INBOX("backup_private_key_to_inbox"),
     ENCRYPT_PRIVATE_KEYS("encrypt_private_keys"),
     LOAD_GMAIL_ALIASES("load_gmail_aliases");
@@ -63,6 +64,25 @@ interface Action : Parcelable {
 
   companion object {
     const val TAG_NAME_ACTION_TYPE = "actionType"
-    const val USER_SYSTEM = "system"
+  }
+
+  data class None(override var id: Long = 0) : Action {
+    override val email: String? = null
+    override val version: Int = 0
+    override val type: Type = Type.NONE
+
+    constructor(parcel: Parcel) : this(parcel.readLong())
+
+    override suspend fun run(context: Context) {}
+    override fun writeToParcel(parcel: Parcel, flags: Int) {
+      parcel.writeLong(id)
+    }
+
+    override fun describeContents(): Int = 0
+
+    companion object CREATOR : Parcelable.Creator<None> {
+      override fun createFromParcel(parcel: Parcel): None = None(parcel)
+      override fun newArray(size: Int): Array<None?> = arrayOfNulls(size)
+    }
   }
 }

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/util/google/gson/ActionJsonDeserializer.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/util/google/gson/ActionJsonDeserializer.kt
@@ -32,7 +32,13 @@ class ActionJsonDeserializer : JsonDeserializer<Action> {
   ): Action {
     val jsonObject = json.asJsonObject
 
-    return when (Action.Type.valueOf(jsonObject.get(Action.TAG_NAME_ACTION_TYPE).asString)) {
+    val enum = try {
+      Action.Type.valueOf(jsonObject.get(Action.TAG_NAME_ACTION_TYPE).asString)
+    } catch (e: IllegalArgumentException) {
+      Action.Type.NONE
+    }
+
+    return when (enum) {
       Action.Type.BACKUP_PRIVATE_KEY_TO_INBOX -> context.deserialize(
         json,
         BackupPrivateKeyToInboxAction::class.java
@@ -47,6 +53,8 @@ class ActionJsonDeserializer : JsonDeserializer<Action> {
         json,
         LoadGmailAliasesAction::class.java
       )
+
+      else -> Action.None()
     }
   }
 }

--- a/FlowCrypt/src/test/java/com/flowcrypt/email/util/google/gson/ActionJsonDeserializerTest.kt
+++ b/FlowCrypt/src/test/java/com/flowcrypt/email/util/google/gson/ActionJsonDeserializerTest.kt
@@ -1,0 +1,37 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.util.google.gson
+
+import com.flowcrypt.email.service.actionqueue.actions.Action
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+/**
+ * @author Denis Bondarenko
+ * Date: 10/27/22
+ * Time: 6:30 PM
+ * E-mail: DenBond7@gmail.com
+ */
+class ActionJsonDeserializerTest {
+
+  @Test
+  fun testDeserializeUnknownTypeAsNone() {
+    val someJson = """
+      {
+         "id":"0",
+         "email":"0",
+         "version":"0",
+         "actionType":"UNKNOWN"
+      }
+    """.trimIndent()
+    val gson: Gson =
+      GsonBuilder().registerTypeAdapter(Action::class.java, ActionJsonDeserializer()).create()
+    val action = gson.fromJson(someJson, Action::class.java)
+    assertEquals(Action.Type.NONE, action.type)
+  }
+}

